### ISSUE TITLE
[ec2] Automatically generated resource spec file for aws-ec2 has the wrong type for field

### ISSUE THREAD

**Issue by cyurio** [aws/aws-cdk#10641]
### The Question
In the autogenerated files for ec2 CfnNetworkInterface has the wrong type for the ipv6Addresses field. When adding ipv6addresses, it will only take a single InstanceIpv6AddressProperty, and if you try to then deploy it fails since cloudformation expects a list.

From my ec2.generated.d.ts file:
      `readonly ipv6Addresses?: CfnNetworkInterface.InstanceIpv6AddressProperty | cdk.IResolvable;`

compare this to the privateipaddresses field, which is correctly generated as a list
      `readonly privateIpAddresses?: Array<CfnNetworkInterface.PrivateIpAddressSpecificationProperty | cdk.IResolvable> | cdk.IResolvable;`

The same in the genereated ec2.generated.js file:
      `errors.collect(cdk.propertyValidator('ipv6Addresses', CfnNetworkInterface_InstanceIpv6AddressPropertyValidator)(properties.ipv6Addresses));`

vs

      `errors.collect(cdk.propertyValidator('privateIpAddresses', cdk.listValidator(CfnNetworkInterface_PrivateIpAddressSpecificationPropertyValidator))(properties.privateIpAddresses));`

### Other information
I'm not familiar with how the files get autogenerated, but when checking the cloudformation spec file https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/cfn-resource-specification.html, it does show the ipv6addresses field is supposed to be a list. So something is going wrong when generating the ipv6Addresses field, since the privateIpAddresses is correctly generated as a list.

**Comment by rix0rrr** [comment]
On my source tree `ipv6Addresses` seems to be correctly generated as a list. What version are you on?

```ts
    /**
     * `AWS::EC2::NetworkInterface.Ipv6Addresses`
     * @see http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-network-interface.html#cfn-ec2-networkinterface-ipv6addresses
     */
    readonly ipv6Addresses?: Array<CfnNetworkInterface.InstanceIpv6AddressProperty | cdk.IResolvable> | cdk.IResolvable;
```

**Comment by rix0rrr** [comment]
Changed in 18.3.0: https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/cfnspec/CHANGELOG.md#cloudformation-resource-specification-v1830

**Comment by rix0rrr** [comment]
Released in 1.64.0: https://github.com/aws/aws-cdk/commit/dbdc7ff20812157be518229ee9be90a5bbcb8d65
